src/puppetlabs/trapperkeeper/services/webrouting/webrouting_service_core.clj

```diff
@@ -123,6 +123,14 @@
         add-servlet-handler (:add-servlet-handler webserver-service)]
     (add-servlet-handler servlet path opts)))
 
+(schema/defn ^:always-validate add-websocket-handler!
+  [context webserver-service
+   svc :- (schema/protocol tk-services/Service)
+   handlers options :- RouteOption]
```

Comment:
I think the options schema needs to stay as what you had before (merging `RouteOption` with the `CommonOptions` from the Jetty9 core namespace, and dissociating the `:server-id` key), as there is an option in the `CommonOptions` in the Jetty9 core namespace that all handlers need to support (`:redirect-if-no-trailing-slash`). The schema can probably just be called `CommonOptions`, and should be used for `add-ring-handler!` as well in place of `RouteOption`.